data_acquisition/py/runGEEperTile.py

```diff
@@ -0,0 +1,500 @@
+  tiles = file.read()
+
+# get EE/Google settings from yml file
+proj = yml['proj'][0]
+proj_folder = yml['proj_folder'][0]
+
+# get/save start date
+yml_start = yml['start_date'][0]
+yml_end = yml['end_date'][0]
+
+# store run date for versioning
+run_date = yml['run_date'][0]
+
+if yml_end == 'today':
+  yml_end = run_date
+
+# gee processing settings
+buffer = yml['site_buffer'][0]
+cloud_filt = yml['cloud_filter'][0]
+cloud_thresh = yml['cloud_thresh'][0]
+
+try: 
+  dswe = yml['DSWE_setting'][0].astype(str)
+except AttributeError: 
+  dswe = yml['DSWE_setting'][0]
+
+# get extent info
+extent = (yml['extent'][0]
+  .split('+'))
+
+if 'site' in extent:
+  locations = read_csv('data_acquisition/in/locs.csv')
+  # convert locations to an eeFeatureCollection
+  locs_feature = csv_to_eeFeat(locations, yml['location_crs'][0])
+
+
+if 'poly' in extent:
+  #if polygon is in extent, check for shapefile
+  shapefile = yml['polygon'][0]
+  # if shapefile provided by user 
+  if shapefile == True:
+    # load the shapefile into a Fiona object
+    with fiona.open('data_acquisition/out/user_polygon.shp') as src:
+      shapes = ([ee.Geometry.Polygon(
+        [[x[0], x[1]] for x in feature['geometry']['coordinates'][0]]
+        ) for feature in src])
+  else: #otherwise use the NHDPlus file
+    # load the shapefile into a Fiona object
+    with fiona.open('data_acquisition/out/NHDPlus_polygon.shp') as src:
+      shapes = ([ee.Geometry.Polygon(
+        [[x[0], x[1]] for x in feature['geometry']['coordinates'][0]]
+        ) for feature in src])
+  # Create an ee.Feature for each shape
+  features = [ee.Feature(shape, {}) for shape in shapes]
+  # Create an ee.FeatureCollection from the ee.Features
+  poly_feat = ee.FeatureCollection(features)
+
+
+if 'center' in extent:
+  if yml['polygon'][0] == True:
+    centers_csv = read_csv('data_acquisition/out/user_polygon_centers.csv')
+    # load the shapefile into a Fiona object
+    centers = csv_to_eeFeat(centers_csv, yml['poly_crs'][0])
+  else: #otherwise use the NHDPlus file
+    centers_csv = read_csv('data_acquisition/out/NHDPlus_polygon_centers.csv')
+    centers = csv_to_eeFeat(centers_csv, 'EPSG:4326')
+  # Create an ee.FeatureCollection from the ee.Features
+  ee_centers = ee.FeatureCollection(centers)    
+
+  
+
+##############################################
+##---- CREATING EE FEATURECOLLECTIONS   ----##
+##############################################
+
+
+wrs = (ee.FeatureCollection('projects/ee-ls-c2-srst/assets/WRS2_descending')
+  .filterMetadata('PR', 'equals', tiles))
+
+wrs_path = int(tiles[:3])
+wrs_row = int(tiles[-3:])
+
+#grab images and apply scaling factors
+l7 = (ee.ImageCollection('LANDSAT/LE07/C02/T1_L2')
+    .filter(ee.Filter.lt('CLOUD_COVER', ee.Number.parse(str(cloud_thresh))))
+    .filterDate(yml_start, yml_end)
+    .filter(ee.Filter.eq('WRS_PATH', wrs_path))
+    .filter(ee.Filter.eq('WRS_ROW', wrs_row)))
```

Comment:
```suggestion
l7 = (ee.ImageCollection('LANDSAT/LE07/C02/T1_L2')
    .filter(ee.Filter.lt('CLOUD_COVER', ee.Number.parse(str(cloud_thresh))))
    .filterDate(yml_start, yml_end) # for user-defined date range
    .filterDate(‘1999-05-28’, ‘2019-12-31’) # for valid dates
    .filter(ee.Filter.eq('WRS_PATH', wrs_path))
    .filter(ee.Filter.eq('WRS_ROW', wrs_row)))
```
	l7 = ee.Imagecollection(‘LANDSAT/LE07/C02/T1_L2’) \
		.filterDate(‘1999-05-28’’, ‘2019-12-31’